application/src/main/java/org/togetherjava/tjbot/commands/moderation/attachment/BlacklistedAttachmentListener.java

```diff
@@ -0,0 +1,115 @@
+package org.togetherjava.tjbot.commands.moderation.attachment;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.MessageBuilder;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.requests.RestAction;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.moderation.ModAuditLogWriter;
+
+import java.awt.*;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+/**
+ * Reacts to blacklisted attachmenents being posted, upon which they are deleted.
+ */
+public class BlacklistedAttachmentListener extends MessageReceiverAdapter {
+    private final ModAuditLogWriter modAuditLogWriter;
+    private final List<String> blacklistedFileExtensions;
+
+    /**
+     * Creates a AttachmentListener to receive all message sent in any channel.
+     *
+     * @param config to find the blacklisted media attachments
+     * @param modAuditLogWriter to inform the mods about the suspicious attachment
+     */
+    public BlacklistedAttachmentListener(@NotNull Config config,
+            @NotNull ModAuditLogWriter modAuditLogWriter) {
+        super(Pattern.compile(".*"));
+        this.modAuditLogWriter = modAuditLogWriter;
+        this.blacklistedFileExtensions = config.getBlacklistedFileExtensions();
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        if (event.getAuthor().isBot() || event.isWebhookMessage()) {
+            return;
+        }
+        if (doesMessageContainBlacklistedContent(event.getMessage())) {
+            handleBadMessage(event.getMessage());
+            warnMods(event.getMessage());
+        }
+    }
+
+    private void handleBadMessage(@NotNull Message message) {
+        message.delete().flatMap(any -> dmUser(message)).queue();
+    }
+
+    private RestAction<Message> dmUser(@NotNull Message message) {
+        Message dmMessage = createDmMessage(message);
+        return message.getJDA()
+            .openPrivateChannelById(message.getAuthor().getIdLong())
+            .flatMap(channel -> channel.sendMessage(dmMessage));
+    }
+
+    private Message createDmMessage(Message originalMessage) {
+        String originalMessageContent = originalMessage.getContentDisplay();
```

Comment:
Dont use `contentDisplay`, use `contentRaw` instead. As long as its posted as embed, pings and the like arent executed anyways.